chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/collector/BasicSessionCollector.java

```diff
@@ -90,19 +99,34 @@ public void onSessionExecuted(String sessionId, String sessionComment) {
     public void onSessionExecuted(String sessionId, String sessionComment, SessionExecutionStatus status) {
         Namespace namespace = Namespace.of(SESSION, sessionId);
         Multimap<String, Object> objectsMap = HashMultimap.create();
-        objectsMap.put(END_TIME, System.currentTimeMillis());
+    
+        sessionInfoProvider.setEndTime(System.currentTimeMillis());
+        objectsMap.put(END_TIME, sessionInfoProvider.getEndTime());
+        
         objectsMap.put(TASK_EXECUTED, taskCounter);
-        Integer failedTasks = taskExecutionStatusProvider.getTaskIdsWithStatus(TaskData.ExecutionStatus.FAILED).size();
+        Integer failedTasks = getTaskCountWithStatus(TaskData.ExecutionStatus.FAILED);
         objectsMap.put(FAILED, failedTasks);
-        if(failedTasks > 0) {
-            if(SessionExecutionStatus.EMPTY.equals(status)){
+        if (failedTasks > 0) {
+            if (SessionExecutionStatus.EMPTY.equals(status)) {
                 status = SessionExecutionStatus.TASK_FAILED;
             }
         }
         objectsMap.put(ERROR_MESSAGE, status.getMessage());
 
         keyValueStorage.putAll(namespace, objectsMap);
     }
+    
+    private int getTaskCountWithStatus(final TaskData.ExecutionStatus status) {
+        int counter = 0;
+        for (Task task : configuration.getTasks()) {
+            if (task instanceof WorkloadTask) {
```

Comment:
I don't know. Maybe originally there were supposed to be some code in between. 
Anyway, it doesn't play any role which way to choose here